sdk/ai/azopenai/autorest.md

```diff
@@ -22,130 +21,219 @@ slice-elements-byval: true
 
 ## Transformations
 
+Fix deployment and endpoint parameters so they show up in the right spots
+
 ``` yaml
 directive:
   # Add x-ms-parameter-location to parameters in x-ms-parameterized-host
-  - from: openapi-document
-    where: $.servers.0.variables.endpoint
+  - from: swagger-document
+    where: $["x-ms-parameterized-host"].parameters.0
     debug: true
     transform: $["x-ms-parameter-location"] = "client";
-
   # Make deploymentId a client parameter
   # This must be done in each operation as the parameter is not defined in the components section
-  - from: openapi-document
+  - from: swagger-document
     where: $.paths..parameters..[?(@.name=='deploymentId')]
     transform: $["x-ms-parameter-location"] = "client";
+```
+
+## Polymorphic adjustments
+
+The polymorphic _input_ models all expose the discriminator but it's ignored when serializing 
```

Comment:
What's the motivation for this, to just reduce the public surface area? At any rate, we have to be sure that for "root" and "sub-root" discriminated types that we export the discriminator. This is to handle cases where a type isn't yet modeled in the SDK but is supported by the service.

Reply:
Ah, okay. I'll bring them back, didn't think about the "expandable future enum" part.

Reply:
It's back by popular demand!

Reply:
You do have a good point though that for the concrete types, exposing the discriminator field seems unnecessary (for inputs) as we automatically populate the value and ignore any user-specified value. On the surface, it also seems unnecessary for output types as well. We could consider making the configurable in code generation, unless we can come up with a reason why it would be bad.

Reply:
But we should not remove it for the "root" type(s) to support the "expandable future enum" thing mentioned earlier.